src/main/java/org/testeditor/fixture/host/net/Connection.java

```diff
@@ -0,0 +1,227 @@
+                try {
+                    Thread.sleep(1000);
+                } catch (final InterruptedException ex) {
+                    if (s3270Process != null) {
+                        s3270Process.destroy();
+                    }
+                    if (s3270Process != null) {
+                        s3270Process.destroy();
+                    }
+                }
+            }
+        }).start();
+    }
+
+    /**
+     *
+     * @return true if connection is established to mainframe through
+     *         s3270/ws3270 emulation and process is still running, false
+     *         otherwise.
+     */
+    public boolean isConnected() {
+        if (s3270Process == null || in == null || out == null) {
+            return false;
+        } else {
+            final Result r = doCommand("");
+            if (r.getStatus().matches(". . . C.*")) {
+                return true;
+            } else {
+                out.println("quit");
+                out.flush();
+                s3270Process.destroy();
+                s3270Process = null;
+                in = null;
+                out = null;
+                return false;
+            }
+        }
+    }
+
+    /**
+     * A s3270 command will be executed with this method. The whole
+     * communication with s3270 will be accessed through this method.
+     */
+    public Result doCommand(final String command) {
+        processAvailable();
+        try {
+            out.println(command);
+            out.flush();
+            logger.debug("*****************************************************************************************");
+            logger.debug("---> Command sent: \"{}\"", command);
+            List<String> lines = new ArrayList<String>();
+            readOutput(lines);
+            int size = lines.size();
+            if (size > 0) {
+                Result result = new Result(lines.subList(0, size - 1), lines.get(size - 1));
+                logger.debug(
+                        "*****************************************************************************************");
+                return result;
+            } else {
+                throw new RuntimeException("no status received in command: " + command);
+            }
+        } catch (final IOException ex) {
+            throw new RuntimeException("IOException during command: " + command, ex);
+        }
+    }
+
+    private void readOutput(List<String> lines) throws IOException {
+        while (true) {
+            String line = in.readLine();
+            if (line == null) {
+                logger.error("Line is null");
+                // if we get here, it's a more obscure error
+                throw new RuntimeException("s3270 process not responding");
+            }
+            logger.debug("<--- {}", line);
+            if (line.equals("ok")) {
+                break;
+            }
+            lines.add(line);
+        }
+    }
+
+    private void processAvailable() {
```

Comment:
name suggests that bool is returned. Maybe use different name (e.g. assertProcessAvailable) ...